imap_processing/codice/make_xtce.py

```diff
@@ -0,0 +1,237 @@
+""" The module will take an Excel file and convert it into an XTCE formatted XML file.
+
+"""
+import xml.etree.ElementTree as ET
+
+import pandas as pd
+
+sheet_name = "P_COD_AUT"  # Assuming the sheet name is correct
+
+# Create CCSDS Header parameters. This should be consistent for all CCSDS packets.
+ccsds_parameters = [
+    {
+        "name": "VERSION",
+        "parameterTypeRef": "uint3",
+        "description": "CCSDS Packet Version Number (always 0)",
+    },
+    {
+        "name": "TYPE",
+        "parameterTypeRef": "uint1",
+        "description": "CCSDS Packet Type Indicator (0=telemetry)",
+    },
+    {
+        "name": "SEC_HDR_FLG",
+        "parameterTypeRef": "uint1",
+        "description": "CCSDS Packet Secondary Header Flag (always 1)",
+    },
+    {
+        "name": "PKT_APID",
+        "parameterTypeRef": "uint11",
+        "description": "CCSDS Packet Application Process ID",
+    },
+    {
+        "name": "SEG_FLGS",
+        "parameterTypeRef": "uint2",
+        "description": "CCSDS Packet Grouping Flags (3=not part of group)",
+    },
+    {
+        "name": "SRC_SEQ_CTR",
+        "parameterTypeRef": "uint14",
+        "description": "CCSDS Packet Sequence Count (increments with each new packet)",
+    },
+    {
+        "name": "PKT_LEN",
+        "parameterTypeRef": "uint16",
+        "description": "CCSDS Packet Length (number of bytes after Packet length minus 1)",
+    },
+    {
```

Comment:
I would take this out from CCSDS header element. I think it's instrument specific. Up until, PKT_LEN is common to all.